platform/migration/platform-migratable-api/src/main/java/org/codice/ddf/migration/MigrationReport.java

```diff
@@ -0,0 +1,219 @@
+    public default Stream<MigrationException> errors() {
+        return messages().filter(MigrationException.class::isInstance)
+                .map(MigrationException.class::cast);
+    }
+
+    /**
+     * Retrieves all warnings recorded by the operation that generated this migration report.
+     *
+     * @return a stream of all recorded warnings (may be empty)
+     */
+    public default Stream<MigrationWarning> warnings() {
+        return messages().filter(MigrationWarning.class::isInstance)
+                .map(MigrationWarning.class::cast);
+    }
+
+    /**
+     * Retrieves all informational messages recorded by the operation that generated this migration
+     * report.
+     *
+     * @return a stream of all recorded info messages (may be empty)
+     */
+    public default Stream<MigrationInformation> infos() {
+        return messages().filter(MigrationInformation.class::isInstance)
+                .map(MigrationInformation.class::cast);
+    }
+
+    /**
+     * Checks if the operation that generated this migration report was successful or not.
+     * <p>
+     * <i>Note:</i> A successful operation might still report warnings.
+     * <p>
+     * Invoking this method will call all registered code via {@link #doAfterCompletion(Consumer)} first.
+     *
+     * @return <code>true</code> if the operation was successfull; <code>false</code> if not
+     */
+    public boolean wasSuccessful();
+
+    /**
+     * Runs the specified code and report whether or not it was successful.
+     * <p>
+     * <i>Note:</i> This method will only account for errors generated from the point where the
+     * provided code is called to the moment it terminates.
+     *
+     * @param code the code to run
+     * @return <code>true</code> if no new errors were recorded while running the provided code;
+     * <code>false</code> if at least one error was recorded
+     */
+    public boolean wasSuccessful(@Nullable Runnable code);
+
+    /**
+     * Runs the specified code and report whether or not it was successful.
+     * <p>
+     * <i>Note:</i> This method will only account for errors generated from the point where the
+     * provided code is called to the moment it terminates.
+     *
+     * @param code the code to run
+     * @return <code>true</code> if no new errors were recorded while running the provided code;
+     * <code>false</code> if at least one error was recorded
+     * @throws IOException if the code executed throws it
+     */
+    public boolean wasIOSuccessful(@Nullable ERunnable<IOException> code) throws IOException;
+
+    /**
+     * Checks if the operation that generated this migration recorded any information messages.
+     * <p>
+     * Invoking this method will call all registered code via {@link #doAfterCompletion(Consumer)} first.
+     *
+     * @return <code>true</code> if the operation recorded at least one informational message;
+     * <code>false</code> if not
+     */
+    public boolean hasInfos();
```

Comment:
I could have provided a default implementation. But since (as you will see when you get to the implementation) I optimized it differently, I didn't see a need to add a default method that was never called.